src/Features/Core/Portable/SplitOrMergeIfStatements/IIfLikeStatementGenerator.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.SplitOrMergeIfStatements
+{
+    /// <summary>
+    /// An if-like statement is either an if statement or an else-if clause.
+    /// When querying the syntax, C# else if chains are "flattened" and modeled to look like VB else-if clauses,
+    /// so an if-like statement can be followed a sequence of else-if clauses (which are themselves if-like statements)
+    /// and an optional final else clause. These else-if clauses are treated as independent when removing or inserting.
+    /// </summary>
+    internal interface IIfLikeStatementGenerator : ILanguageService
+    {
+        bool IsIfLikeStatement(SyntaxNode node);
+
+        bool IsCondition(SyntaxNode expression, out SyntaxNode ifLikeStatement);
+
+        bool IsElseIfClause(SyntaxNode node, out SyntaxNode parentIfLikeStatement);
```

Comment:
does this get the first If-Statement of the chain?  Or just the direct parent?  *should* it get the first?

Reply:
No, it should not get the first.